adafruit-libraries: update all subpackages

InuSasha opened the pull request:
- ~~switch to sha in full length~~
- add script for automatic update
- fix build of DHT package

CvH commented:
@InuSasha to avoid the the git user/pw you can use the "public api" - I had this hacky line in stash to ask for the last commit date from git (I am pretty sure this could be simplified a lot)
```
# get last commit date
GIT_REV=$( curl -s https://api.github.com/repos/LibreELEC/LibreELEC.tv/commits/master | tac | tac | head -8 | grep date | awk '{ print $2 }' | sed s/\"//g | sed s/,//g | sed 's/.\{10\}$//');
```
`switch to sha in full length` do we want this ?

InuSasha commented:
- replaced copyright
- better help text

@CvH 
I used the full sha, because i don't know how github shrink the full. and there is no big human interaction. So it is the easiest way, i think.

The github login is only needed, if we hit the rate limit (after 4 runs of the script, for example. or in combination with other scripts). I changed the help text, to make it more obvious.

chewitt commented:
@InuSasha the short githash is the first 7 characters of the full hash

InuSasha commented:
@CvH switched back to short sha
@chewitt thanks

chewitt commented:
@InuSasha If you import from ~/.gitconfig (as suggested by milhouse in Slack) the maintenance script can be simplified.

CvH commented:
@InuSasha https://github.com/LibreELEC/LibreELEC.tv/blob/master/tools/distro-tool#L44-L45 was milhouse suggestion

InuSasha commented:
switched to milhouse solution, for the github credentials

MilhouseVH commented:
It's ~/.git.conf:
```
GIT_USERNAME="username"
GIT_PASSWORD="password_or_token"
```

MilhouseVH commented on tools/mkpkg/update_adafruit-libraries:
```diff
@@ -0,0 +1,97 @@
+  fi
+  
+  # compare hash
+  GIT_HASH=$(cut -c -7 <<< $GIT_HASH)
```
`GIT_HASH=${GIT_HASH:0:7}` is easier.

MilhouseVH commented on tools/mkpkg/update_adafruit-libraries:
```diff
@@ -0,0 +1,97 @@
+  fi
+
+  # replace old with new hash
+  sed "s|^PKG_VERSION=.*$|PKG_VERSION=\"${GIT_HASH}\"|" \
```
`sed "s|^PKG_VERSION=.*$|PKG_VERSION=\"${GIT_HASH}\"|" -i ${PKG_DIR}/package.mk`
then you can drop the following 3 lines (72-74).

MilhouseVH commented on tools/mkpkg/update_adafruit-libraries:
```diff
@@ -0,0 +1,97 @@
+  # revision
+  REV=$(( $PKG_REV + 1 ))
+  echo "increase revision $PKG_REV -> $REV"
+  sed "s|^PKG_REV=.*$|PKG_REV=\"${REV}\"|" \
```
Same as above.

MilhouseVH commented on tools/mkpkg/update_adafruit-libraries:
```diff
@@ -0,0 +1,97 @@
+  
+  # get new hash from master branch
+  URL=$(sed "s|\(www\.\)*github.com/|api.github.com/repos/|;s|/archive/.*|/commits/master|" <<< $PKG_URL)
+  GIT_HASH=$(curl --silent --header "Accept: application/vnd.github.VERSION.sha" $CURL_LOGIN $URL)
```
`VERSION` should be `v3` rather than the literal VERSION: https://developer.github.com/v3/media/

InuSasha commented:
implement @MilhouseVH suggestions, and feel like a rookie :)

InuSasha commented:
add SHA256 hashes to packages and update script
